simpleline/render/in_out_manager.py

```diff
@@ -0,0 +1,241 @@
+        return input_queue.get()  # return the user input
+
+    def _thread_input(self, queue_instance, prompt, hidden):
+        """This method is responsible for interruptable user input.
+
+        It is expected to be used in a thread started on demand
+        and returns the input via the communication Queue.
+
+        :param queue_instance: communication queue_instance to be used
+        :type queue_instance: queue.Queue instance
+
+        :param prompt: prompt to be displayed
+        :type prompt: Prompt instance
+
+        :param hidden: whether typed characters should be echoed or not
+        :type hidden: bool
+        """
+        if hidden:
+            getpass_func = self.getpass
+            data = getpass_func(prompt)
+        else:
+            widget = TextWidget(str(prompt))
+            widget.render(self._width)
+            lines = widget.get_lines()
+            sys.stdout.write("\n".join(lines) + " ")
+            sys.stdout.flush()
+            if not RAW_INPUT_LOCK.acquire(False):
+                # raw_input is already running
+                return
+            else:
+                # lock acquired, we can run input
+                try:
+                    data = self._get_input()
+                except EOFError:
+                    data = ""
+                finally:
+                    RAW_INPUT_LOCK.release()
+
+        queue_instance.put(data)
+
+    def _get_input(self):
+        return input()
+
+    def _process_input(self, active_screen, key):
+        """Method called internally to process unhandled input key presses.
+
+        :param active_screen: Screen for input processing.
+        :type active_screen: Class based on `simpleline.render.screen_stack.screen_data`.
+
+        :param key: The string entered by user.
+        :type key: String.
+
+        :return: Return data object with result status and state.
+        :rtype: `simpleline.render.in_out_manager.UserInputResult` class.
+        """
+        # delegate the handling to active screen first
+        try:
+            key = active_screen.ui_screen.input(active_screen.args, key)
+            if key == INPUT_PROCESSED:
+                return UserInputResult(True)
+            elif key == INPUT_DISCARDED:
+                return UserInputResult(False)
+        except ExitMainLoop:
+            raise
+        except Exception:    # pylint: disable=broad-except
+            self._event_loop.enqueue_signal(ExceptionSignal(self))
+            return UserInputResult(False)
+
+        # global refresh command
+        if key == Prompt.REFRESH:
+            return UserInputResult(True, UserInputStateEnum.REFRESH)
+
+        # global close command
+        if key == Prompt.CONTINUE:
+            return UserInputResult(True, UserInputStateEnum.CONTINUE)
+
+        # global quit command
+        if key == Prompt.QUIT:
+            return UserInputResult(True, UserInputStateEnum.QUIT)
+
+        return UserInputResult(False)
```

Comment:
True, I will remove the `UserInputResult` class and add a new method `was_successful()`to `UserInputStateEnum`. 

Thanks for improvement.